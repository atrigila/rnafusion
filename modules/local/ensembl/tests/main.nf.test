nextflow_process {

    name "Test Process ENSEMBL_DOWNLOAD"
    script "../main.nf"
    process "ENSEMBL_DOWNLOAD"

    test("Should download Ensembl files") {

        when {
            params {
                outdir = "tests/results"
                genome = "GRCh38"
            }
            process {
                """
                input[0] = "102"
                input[1] = "GRCh38"
                input[2] = "GRCh38"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.gtf[0][1]).name,
                    file(process.out.primary_assembly[0][1]).name,
                    file(process.out.chrgtf[0][1]).name,
                    process.out.versions
                ).match() }
            )
        }

    }

}
