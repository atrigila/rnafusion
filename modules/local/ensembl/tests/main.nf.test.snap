{
    "Should download Ensembl files": {
        "content": [
            "Homo_sapiens.GRCh38.102.gtf",
            "Homo_sapiens.GRCh38.102.dna.primary_assembly.fa",
            "Homo_sapiens.GRCh38.102.chr.gtf",
            [
                "versions.yml:md5,efb55665a151fe2b4d9eac70c0e01bb0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T14:06:24.319131837"
    }
}